### PR TITLE
[FLINK-34505][table] Migrate WindowGroupReorderRule to java.

### PR DESCRIPTION
**Pull request by liuyongvs**
## What is the purpose of the change

The PR migrates WindowGroupReorderRule to java
it doesn't touch WindowGroupReorderRuleTest to be sure that java version continues passing it

## Verifying this change

This change is already covered by existing tests

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/rules/logical/WindowGroupReorderRule.java [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+                            .map(
+                                    index ->
+                                            new RexInputRef(
+                                                    index, newFieldList.get(index).getValue()))
+                            .collect(Collectors.toList());
+
+            LogicalProject project =
+                    LogicalProject.create(
+                            newLogicalWindow,
+                            java.util.Collections.emptyList(),
```
Comment (snuyanzin) [inline flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/rules/logical/WindowGroupReorderRule.java:136]: why can't we have `Collections.emptyList()` here?

File: flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/rules/logical/WindowGroupReorderRule.java [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+                            window.getCluster().getPlanner().emptyTraitSet(),
+                            input,
+                            window.constants,
+                            intermediateRowType,
+                            sequenceGroups);
+
+            List<Integer> sortedIndices =
+                    IntStream.range(0, mapToOldTypeIndexes.size())
+                            .boxed()
+                            .sorted(Comparator.comparingInt(i -> mapToOldTypeIndexes.get(i)))
```
Comment (snuyanzin) [inline flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/rules/logical/WindowGroupReorderRule.java:122]:
```suggestion
                            .sorted(Comparator.comparingInt(mapToOldTypeIndexes::get))
```

File: flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/rules/logical/WindowGroupReorderRule.java (hunk 1 above)
Comment (liuyongvs) [reply flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/rules/logical/WindowGroupReorderRule.java:136]: done @snuyanzin

**Comment by snuyanzin** [comment]
@liuyongvs the ci still fails could you please have a look here?

**Comment by snuyanzin** [comment]
still failing on ci

**Comment by liuyongvs** [comment]
hi @snuyanzin ci passed, will you look again?

**Comment by liuyongvs** [comment]
hi @snuyanzin ci passed, will you look again?

**Review by snuyanzin — APPROVED** [review]

**Pull request merged**